src/models/kv_cache.cpp

```diff
@@ -113,6 +113,12 @@ void KV_Cache_Combined::PickPastState(std::span<const int32_t> beam_indices, int
   }
 }
 
+bool KV_Cache::IsCacheNeeded(const Model& model) {
+  char past_key_name[64];
+  snprintf(past_key_name, std::size(past_key_name), model.config_->model.decoder.inputs.past_key_names.c_str(), 0);
```

Comment:
should we check the return value to see if the buffer was actually big enough? also noticed a few other similar calls to `snprintf` in this file.

Reply:
Made changes to address this.